lib/rack/test.rb

```diff
@@ -230,24 +202,30 @@ def follow_redirect!
 
       private
 
+      # Normalize URI based on given URI/path and environment.
       def parse_uri(path, env)
-        URI.parse(path).tap do |uri|
-          uri.path = "/#{uri.path}" unless uri.path[0] == '/'
-          uri.host ||= @default_host
-          uri.scheme ||= 'https' if env['HTTPS'] == 'on'
-        end
+        uri = URI.parse(path)
+        uri.path = "/#{uri.path}" unless uri.path.start_with?('/')
+        uri.host ||= @rack_mock_session.default_host
+        uri.scheme ||= 'https' if env['HTTPS'] == 'on'
+        uri
       end
 
+      DEFAULT_ENV = {
+        'rack.test' => true,
+        'REMOTE_ADDR' => '127.0.0.1',
+        'SERVER_PROTOCOL' => 'HTTP/1.0',
+        'HTTP_VERSION' => 'HTTP/1.0'
```

Comment:
Should we be encouraging use of `HTTP_VERSION` here?

Reply:
It's needed for backwards compatibility with older versions of Rack.  If you want, we can stop setting this if the rack version is use is rack 3+ (since SERVER_PROTOCOL should always be used in that case based on rack SPEC).

Reply:
I think it might be a good idea to do something like that.